Add autodocs page to documentation

lkdvos opened the pull request:
Here I added the setup for including a page with all docstrings at the end of the documentation. As I would assume this is something that most packages will want to do anyways, I think it might be convenient to immediately set that up in the skeleton.

lkdvos commented:
Agreed. (I made everything "Library") right now, but I honestly would prefer to call this "Index", but that conflicts with the `index.md` we have set up right now (which I would maybe rename to `home.md` instead?). I'm definitely not hung up on this, just something to consider.

mtfishman commented:
> Agreed. As a side note, I made everything "Library" right now, but I honestly would prefer to call this "Index", but that conflicts with the `index.md` we have set up right now (which I would maybe rename to `home.md` instead?). I'm definitely not hung up on this, just something to consider.

I'd be good with `["Home" => "home.md", "Index" => "index.md"]` (as long as `"index.md"` isn't treated as special in some other way by Documenter.jl?).

lkdvos commented:
Oh you are right, indeed Documenter.jl treats that as the landing page, I did not consider that. In that case, it's probably ok as is.

mtfishman commented:
What do you think of "Reference" as opposed to "Library"?

lkdvos commented:
I renamed and rebased on the latest changes. I can still bump to v0.2.2 if you prefer, but probably here it's okay to have both changes in a single patch

mtfishman commented:
> here it's okay to have both changes in a single patch

Agreed.
